engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeCallableNode.java

```diff
@@ -251,6 +257,34 @@ public Object invokeDynamicSymbol(
     }
   }
 
+  @Specialization(guards = "warnings.hasWarnings(warning)")
+  public Object invokeWarnings(
+      Object warning,
+      VirtualFrame callerFrame,
+      State state,
+      Object[] arguments,
+      @Cached InvokeCallableNode invokeCallableNode,
+      @CachedLibrary(limit = "3") WarningsLibrary warnings) {
+    try {
+      var result = invokeCallableNode.execute(
+                  warnings.removeWarnings(warning),
+                  callerFrame,
+                  state,
+                  arguments);
+
+      Warning[] extracted = warnings.getWarnings(warning, null);
+      if (result instanceof DataflowError) {
+        return result;
+      } else if (result instanceof WithWarnings withWarnings) {
+        return withWarnings.prepend(extracted);
+      } else {
+        return new WithWarnings(result, extracted);
+      }
```

Comment:
This feels like an universal piece of logic when we are attaching warnings to a result which may already also contain new warnings.

Is similar logic used elsewhere?
Maybe we can refactor it into a separate function `WithWarnings::attachToObject`?